.github/workflows/test.yml

```diff
@@ -34,8 +40,17 @@ jobs:
         with:
           node-version: ${{ matrix.node-version }}
 
-      - name: Install specific npm version
-        run: npm install -g npm@^9.6.7
+      - name: Install npm version matching NodeJS
+        if: ${{ matrix.node-version == vars.WF_NODEJS_LATEST }}
+        run: npm install -g npm@${{ vars.WF_NPM_LATEST }}
+
+      - name: Install npm version matching NodeJS
+        if: ${{ matrix.node-version == vars.WF_NODEJS_OLD }}
+        run: npm install -g npm@${{ vars.WF_NPM_OLD }}
+
+      - name: Ensure pip is installed
+        if: runner.os == 'macOS'
+        run: python3 -m ensurepip
 
       - name: Install specific python dependencies
         run: pip install setuptools
```

Comment:
I guess on macOc it should be pip3 but last step ensure both pip & setuptools are there so I suggest:
`if: runner.os != 'macOS'`
`run: pip install setuptools`

Reply:
- what do you mean "last step ensure both pip & setuptools"?
- if I target runner.os != 'macOS', setuptools will not be installed on Mac
- the issue is that pip "is" installed but apparently not in PATH I guess (/opt/homebrew/lib/python3.11/site-packages) ; I'm gonna try using "python -m pip install" which is what you're supposed to do anyway

Reply:
I mean that on masos 14 both pip and setuptools are installed at [prevous step `ensurepip` ](https://github.com/Open-Lens/lens/actions/runs/8380636534/job/22950304776#step:7:6)

Reply:
yes, well, I add the ensurepip step because there was an error with on macos but didn't do much digging beyond that. I'll just remove the setuptools step entirely

Reply:
Please no setuptools are only there on macos 14 !